### ISSUE TITLE
InnerAssignmentCheck failed for one line code in Java 14 switch expression

### ISSUE THREAD

**Issue by HaelC** [checkstyle/checkstyle#13086]
I have read check documentation: https://checkstyle.org/config_coding.html#InnerAssignment
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac TestInnerAssignmentCheck.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="InnerAssignment"/>
  </module>
</module>

/var/tmp $ cat TestInnerAssignmentCheck.java
public class TestInnerAssignmentCheck {
    private boolean flag;

    void getOperation(final String operation) {
        switch (operation) {
            case "Y" -> flag = true;  // violation 
            case "N" -> {
                flag = false;
            }
            default -> throw new UnsupportedOperationException();
        }
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.11.0-all.jar -c config.xml TestInnerAssignmentCheck.java
Starting audit...
[ERROR] /Users/hael/Code/Learn/Tools/checkstyle/TestInnerAssignmentCheck.java:6:30:
     Inner assignments should be avoided. [InnerAssignment]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

Checkstyle behaves differently between one line code and code block for Java 14's Switch Expression. In the given code above, checkstyle will complain about `case "Y" -> flag = true;` but not `case "N" -> {flag = false;}`. I'm afraid that it's because `flag = true;` is equivalent to `return flag = true;`? But the return value is not used here in the switch expression, we just want to do the value assignment. 

Expected: no violations on such code.

**Comment by Andromeda227799** [comment]
I am on It

**Comment by AayushSaini101** [comment]
@romani Can i take this

**Comment by nrmancuso** [comment]
@AayushSaini101 yes, take a look at the PR I closed at #13090

**Comment by mahfouz72** [comment]
@romani @nrmancuso there is an open PR for this but no activity since jan.idk if it is abandoned.so can i take this or should we wait

**Comment by nrmancuso** [comment]
@mahfouz72 you can pick it up, I closed the PR
